server/src/controllers/cart.controller.ts

```diff
@@ -0,0 +1,58 @@
+import express from "express";
+import Cart from "../models/Cart/Cart";
+
+const cartController = {
+  async showCart(
+    req: express.Request,
+    res: express.Response
+  ): Promise<express.Response | void> {
+    try {
+      const userId = req.body.user._id;
+      const myCart = await Cart.findById({ userId });
+      return res.status(201).json({ myCart });
+    } catch (err) {
+      return res.status(500).json({
+        error: err.message
+      });
+    }
+  },
+
+  async addToCart(
+    req: express.Request,
+    res: express.Response
+  ): Promise<express.Response | void> {
+    try {
+      const { productId } = req.params;
+      const userId = req.body.user._id;
+      const myCart = (await Cart.findById(userId)) || new Cart();
```

Comment:
The `Cart.findById(userId)` will not find a Cart anytime, because Cart will have it's own cartID. You can pass the cartID in req.body and access it here.